### ISSUE TITLE
NullPointerException with a POST request in a Zest script

### ISSUE THREAD

**Issue by thc202** [zaproxy/zaproxy#2859]
Steps to reproduce the issue:
1. Run ZAP with Zest add-on;
2. Run the following stand alone Zest script:

``` JSON
{
  "about": "This is a Zest script. For more details about Zest visit https://developer.mozilla.org/en-US/docs/Zest",
  "zestVersion": "0.3",
  "title": "X",
  "description": "",
  "prefix": "",
  "type": "StandAlone",
  "parameters": {
    "tokenStart": "{{",
    "tokenEnd": "}}",
    "tokens": {},
    "elementType": "ZestVariables"
  },
  "statements": [
    {
      "url": "http://example.com/",
      "data": "",
      "method": "POST",
      "headers": "",
      "assertions": [],
      "followRedirects": true,
      "cookies": [],
      "index": 1,
      "enabled": true,
      "elementType": "ZestRequest"
    }
  ],
  "authentication": [],
  "index": 0,
  "enabled": true,
  "elementType": "ZestScript"
}
```

3.Note that in the output panel it shows:

```
java.lang.NullPointerExceptionjava.lang.NullPointerException
```

Remarks:
Using the Zest add-on and Zest library with debug info it shows the following stack trace:

```
1842353 [ZAP-ScriptExecutor-X] ERROR org.zaproxy.zap.extension.scripts.OutputPanel  - 
java.lang.NullPointerException
    at org.apache.commons.httpclient.HttpMethodBase.setRequestHeader(HttpMethodBase.java:512)
    at org.mozilla.zest.impl.ZestBasicRunner.send(ZestBasicRunner.java:487)
    at org.mozilla.zest.impl.ZestBasicRunner.send(ZestBasicRunner.java:365)
    at org.mozilla.zest.impl.ZestBasicRunner.runStatement(ZestBasicRunner.java:170)
    at org.zaproxy.zap.extension.zest.ZestZapRunner.runStatement(ZestZapRunner.java:254)
    at org.mozilla.zest.impl.ZestBasicRunner.run(ZestBasicRunner.java:142)
    at org.zaproxy.zap.extension.zest.ZestZapRunner.run(ZestZapRunner.java:141)
    at org.mozilla.zest.impl.ZestBasicRunner.run(ZestBasicRunner.java:98)
    at org.zaproxy.zap.extension.zest.ZestZapRunner.run(ZestZapRunner.java:126)
    at org.mozilla.zest.impl.ZestBasicRunner.runScript(ZestBasicRunner.java:435)
    at org.mozilla.zest.impl.ZestScriptEngine.eval(ZestScriptEngine.java:49)
    at javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
    at org.zaproxy.zap.extension.script.ExtensionScript.invokeScriptImpl(ExtensionScript.java:1037)
    at org.zaproxy.zap.extension.script.ExtensionScript.invokeScript(ExtensionScript.java:976)
    at org.zaproxy.zap.extension.scripts.ConsolePanel$ScriptExecutorThread.run(Unknown Source)
```

For reference:
https://github.com/mozilla/zest/blob/8bb7d055a030eb50cde33afe7ba5130d1b74d7d0/src/org/mozilla/zest/impl/ZestBasicRunner.java#L480-L487

ZAP version:
Version 2.5.0

Add-on:
"Zest - Graphical Security Scripting Language" version 22

**Comment by thc202** [comment]
Maybe this issue should also be open in https://github.com/mozilla/zest?

**Comment by thc202** [comment]
This was fixed in the Zest library mozilla/zest#77 (which needs to be updated in Zest add-on once the new version is available).

**Comment by thc202** [comment]
Fixed in zaproxy/zap-extensions#985.
